src/Appwrite/Utopia/Response/Model/AttributeList.php

```diff
@@ -0,0 +1,62 @@
+<?php
+
+namespace Appwrite\Utopia\Response\Model;
+
+use Appwrite\Utopia\Response;
+use Appwrite\Utopia\Response\Model;
+use Utopia\Database\Document;
+
+class AttributeList extends Model
+{
+    public function __construct()
+    {
+        $this
+            ->addRule('sum', [
+                'type' => self::TYPE_INTEGER,
+                'description' => 'Total sum of items in the list.',
+                'default' => 0,
+                'example' => 5,
+            ])
+            ->addRule('attributes', [
+                'type' => Response::MODEL_ATTRIBUTE,
+                'description' => 'List of attributes.',
+                'default' => [],
+                'array' => true,
+                'getNestedType' => function(Document $attribute) {
+                    return match($attribute->getAttribute('type')) {
+                        self::TYPE_BOOLEAN => Response::MODEL_ATTRIBUTE_BOOLEAN,
+                        self::TYPE_INTEGER => Response::MODEL_ATTRIBUTE_INTEGER,
+                        self::TYPE_FLOAT => Response::MODEL_ATTRIBUTE_FLOAT,
+                        self::TYPE_STRING => match($attribute->getAttribute('format')) {
+                            APP_DATABASE_ATTRIBUTE_EMAIL => Response::MODEL_ATTRIBUTE_EMAIL,
+                            APP_DATABASE_ATTRIBUTE_IP => Response::MODEL_ATTRIBUTE_IP,
+                            APP_DATABASE_ATTRIBUTE_URL => Response::MODEL_ATTRIBUTE_URL,
+                            default => Response::MODEL_ATTRIBUTE_STRING,
+                        },
+                        default => Response::MODEL_ATTRIBUTE,
+                    };
+                },
+            ])
```

Comment:
In terms of syntax, it would be awesome to solve it with something like this:

```php
->addRule('attributes', [
    'type' => [
        Response::MODEL_ATTRIBUTE_BOOLEAN,
        Response::MODEL_ATTRIBUTE_INTEGER,
        Response::MODEL_ATTRIBUTE_FLOAT,
        Response::MODEL_ATTRIBUTE_STRING,
        Response::MODEL_ATTRIBUTE_EMAIL,
        Response::MODEL_ATTRIBUTE_URL,
        Response::MODEL_ATTRIBUTE_IP
    ],
    'description' => 'List of attributes.',
    'default' => [],
    'array' => true,
])
```

Since we need to reflect it later on the Swagger Specifications as well. There we need all possible models as well.

But for this to work, we need to have an identifier present in each Mode.

Reply:
Agreed, we need some attribute to parse the response models correctly later in SDKs

Reply:
Fixed via https://github.com/appwrite/appwrite/pull/1624